### ISSUE TITLE
[Error Bug] SELECT <table> FROM JOIN tables ON (CAST(<decimal> AS MONEY)) Brings Errors

### ISSUE THREAD

**Issue by qwebug** [heterodb/pg-strom#627]
### Describe:

Using SELECT \<table> FROM RIGHT OUTER JOIN 2 tables ON (CAST(\<decimal> AS MONEY)  IN CAST(\<decimal> AS MONEY)) will report an error, when pg\_strom.enabled is turned on.

However it is able to output result, when pg\_strom.enabled is turned off.

#### SQL < pg\_strom.enabled ON >:

```sql
CREATE TABLE t1(c0 INT);
CREATE TABLE t2(LIKE t1);
INSERT INTO t2(c0) VALUES(1);
INSERT INTO t1(c0) VALUES(1);
CREATE SCHEMA extensions;
CREATE EXTENSION pg_strom WITH SCHEMA extensions;
set pg_strom.enabled=on;
SELECT t2 FROM t1 RIGHT OUTER JOIN t2 ON ((CAST(0.1 AS MONEY)) IN (CAST(0.1 AS MONEY)));
```

Result:

```bash
ERROR:  cache lookup failed for attribute 0 of relation 203291
```

#### SQL < pg\_strom.enabled OFF >:

```sql
set pg_strom.enabled=off;
SELECT t2 FROM t1 RIGHT OUTER JOIN t2 ON ((CAST(0.1 AS MONEY)) IN (CAST(0.1 AS MONEY)));
```

Result:

```bash
 t2  
-----
 (1)
(1 row)
```

Futhermore, if change RIGHT OUTER JOIN to LEFT OUTER JOIN or INNER JOIN, errors also occur while pg\_strom.enabled is turned on.

### Environment:

Pg-strom Version: commit 4ac6d95339494f901b93acb384186415dfe63c1b

PostgreSQL Version: 15.3 on x86\_64-pc-linux-gnu, compiled by gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-18), 64-bit

CUDA Version: 12.0.1

NVIDIA Driver Version: 535.86.05

**Comment by kaigai** [comment]
`0786a13d3d59b17f4d110e2ab76942d6ebb4d4ee` should fix this (potential) problem.
Even though the fix at #626 eliminated the code path for this issue, this problem may happen potentially.
So, this patch fixed the case when `custom_scan_tlist` needs a var-node of whole-row-reference.

Thanks for your reporting.

**Comment by qwebug** [comment]
This problem is fixed, after my verification.
Thanks for your work.
